llvm/test/Instrumentation/sanitizers-naked.ll

```diff
@@ -0,0 +1,90 @@
+; NOTE: Assertions have been autogenerated by utils/update_test_checks.py UTC_ARGS: --version 5
+; RUN: opt < %s -passes=asan -S | FileCheck %s --check-prefixes=CHECK-ASAN
+; RUN: opt < %s -passes=tsan -S | FileCheck %s --check-prefixes=CHECK-TSAN
+; RUN: opt < %s -passes=msan -S | FileCheck %s --check-prefixes=CHECK-MSAN
+; RUN: opt < %s -passes=hwasan -S | FileCheck %s --check-prefixes=CHECK-HWASAN
+; RUN: opt < %s -passes=dfsan -dfsan-track-origins=1  -S | FileCheck %s --check-prefixes=CHECK-DFSAN
+; RUN: opt < %s -passes='module(sancov-module)' -sanitizer-coverage-level=3 -sanitizer-coverage-control-flow -S | FileCheck %s --check-prefixes=CHECK-SANCOV
+; RUN: opt < %s -passes='module(sanmd-module)' -sanitizer-metadata-atomics -S | FileCheck %s --check-prefixes=CHECK-SANMD
+
+target datalayout = "e-m:e-i64:64-f80:128-n8:16:32:64-S128"
+target triple = "x86_64-unknown-linux-gnu"
+
+define void @naked_function() naked {
```

Comment:
Can you please add a byval argument to these functions? I think that will make it more likely that something actually gets instrumented and results in a verifier error.

Reply:
May be missing something, how is the argument supposed to lead that something gets instrumented if the argument cannot be used?

Reply:
See https://llvm.godbolt.org/z/K4WdqTdve. Without the byval argument there is nothing to instrument anyway, so you're not really testing anything.

Reply:
Doesn't your function miss naked attribute? The verifier would lead to an error in that case.

Reply:
Yes, that's the point. You want a test case that leads to a verifier error before your change and no longer causes one after your change.

Reply:
OK I see it now, thanks. Should be better. I was slightly confused by the naked_function without being naked.